Error while building the image

I've tried the command `docker compose up` and I get this:

```
#0 9.436 Downloading sentinels-1.1.1-py3-none-any.whl (3.7 kB)
#0 9.457 Downloading wcwidth-0.2.14-py2.py3-none-any.whl (37 kB)
#0 9.532 Building wheels for collected packages: cyvcf2, vcftoolbox
#0 9.533   DEPRECATION: Building 'cyvcf2' using the legacy setup.py bdist_wheel mechanism, which will be removed in a future version. pip 25.3 will enforce this behaviour change. A possible replacement is to use the standardized build interface by setting the `--use-pep517` option, (possibly combined with `--no-build-isolation`), or adding a `pyproject.toml` file to the source tree of 'cyvcf2'. Discussion can be found at https://github.com/pypa/pip/issues/6334
#0 9.533   Building wheel for cyvcf2 (setup.py): started
#0 10.61   Building wheel for cyvcf2 (setup.py): finished with status 'error'
#0 10.61   error: subprocess-exited-with-error
#0 10.61
#0 10.61   × python setup.py bdist_wheel did not run successfully.
#0 10.61   │ exit code: 1
#0 10.61   ╰─> [105 lines of output]
#0 10.61       running bdist_wheel
#0 10.61       running build
#0 10.61       running build_py
#0 10.61       creating build
#0 10.61       creating build/lib.linux-aarch64-3.9
#0 10.61       creating build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/cli.py -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/__init__.py -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/__main__.py -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       creating build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test_cli.py -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test_hemi.py -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test_reader.py -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test_writer.py -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/__init__.py -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       running egg_info
#0 10.61       writing cyvcf2.egg-info/PKG-INFO
#0 10.61       writing dependency_links to cyvcf2.egg-info/dependency_links.txt
#0 10.61       writing entry points to cyvcf2.egg-info/entry_points.txt
#0 10.61       writing requirements to cyvcf2.egg-info/requires.txt
#0 10.61       writing top-level names to cyvcf2.egg-info/top_level.txt
#0 10.61       reading manifest file 'cyvcf2.egg-info/SOURCES.txt'
#0 10.61       reading manifest template 'MANIFEST.in'
#0 10.61       adding license file 'LICENSE'
#0 10.61       writing manifest file 'cyvcf2.egg-info/SOURCES.txt'
#0 10.61       copying cyvcf2/cyvcf2.c -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/cyvcf2.pxd -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/cyvcf2.pyx -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/helpers.c -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/helpers.h -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/relatedness.h -> build/lib.linux-aarch64-3.9/cyvcf2
#0 10.61       copying cyvcf2/tests/bug.vcf.gz -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/decomposed.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/empty.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/issue_198.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/issue_44.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/no-seq-len.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/no-seq-names.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/o.vcf.gz -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/seg.vcf.gz -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-alt-repr.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-diff.csi -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-format-string.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-haploidX.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-hemi.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-invalid-header.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-multiallelic-homozygous-alt.vcf.gz -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-multiallelic-homozygous-alt.vcf.gz.tbi -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-strict-gt-option-flag.vcf.gz -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test-strict-gt-option-flag.vcf.gz.tbi -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.bug.117.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.comp_het.3.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.isa.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.mnp.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.snpeff.bcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.snpeff.bcf.csi -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.snpeff.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.vcf.gz -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test.vcf.gz.tbi -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test_gt_alt_freqs.vcf -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       copying cyvcf2/tests/test_gt_bases.vcf.gz -> build/lib.linux-aarch64-3.9/cyvcf2/tests
#0 10.61       running build_ext
#0 10.61       building 'cyvcf2.cyvcf2' extension
#0 10.61       creating build/temp.linux-aarch64-3.9
#0 10.61       creating build/temp.linux-aarch64-3.9/cyvcf2
#0 10.61       creating build/temp.linux-aarch64-3.9/htslib
#0 10.61       creating build/temp.linux-aarch64-3.9/htslib/cram
#0 10.61       gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -Ihtslib -Icyvcf2 -I/tmp/pip-install-6behzz87/cyvcf2_38e54a24a3b64bd59984ec9faf3d49ac/.eggs/numpy-2.0.2-py3.9-linux-aarch64.egg/numpy/_core/include -I/usr/local/include/python3.9 -c cyvcf2/cyvcf2.c -o build/temp.linux-aarch64-3.9/cyvcf2/cyvcf2.o -Wno-sign-compare -Wno-unused-function -Wno-strict-prototypes -Wno-unused-result -Wno-discarded-qualifiers
#0 10.61       In file included from /tmp/pip-install-6behzz87/cyvcf2_38e54a24a3b64bd59984ec9faf3d49ac/.eggs/numpy-2.0.2-py3.9-linux-aarch64.egg/numpy/_core/include/numpy/ndarraytypes.h:1909,
#0 10.61                        from /tmp/pip-install-6behzz87/cyvcf2_38e54a24a3b64bd59984ec9faf3d49ac/.eggs/numpy-2.0.2-py3.9-linux-aarch64.egg/numpy/_core/include/numpy/ndarrayobject.h:12,
#0 10.61                        from /tmp/pip-install-6behzz87/cyvcf2_38e54a24a3b64bd59984ec9faf3d49ac/.eggs/numpy-2.0.2-py3.9-linux-aarch64.egg/numpy/_core/include/numpy/arrayobject.h:5,
#0 10.61                        from cyvcf2/cyvcf2.c:711:
#0 10.61       /tmp/pip-install-6behzz87/cyvcf2_38e54a24a3b64bd59984ec9faf3d49ac/.eggs/numpy-2.0.2-py3.9-linux-aarch64.egg/numpy/_core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
#0 10.61          17 | #warning "Using deprecated NumPy API, disable it with " \
#0 10.61             |  ^~~~~~~
#0 10.61       cyvcf2/cyvcf2.c: In function ‘__pyx_pf_6cyvcf2_6cyvcf2_3VCF_8seqnames___get__’:
#0 10.61       cyvcf2/cyvcf2.c:14475:18: warning: assignment to ‘char **’ from incompatible pointer type ‘const char **’ [-Wincompatible-pointer-types]
#0 10.61       14475 |   __pyx_v_cnames = bcf_hdr_seqnames(__pyx_v_self->hdr, (&__pyx_v_n));
#0 10.61             |                  ^
#0 10.61       cyvcf2/cyvcf2.c:14584:22: warning: assignment to ‘char **’ from incompatible pointer type ‘const char **’ [-Wincompatible-pointer-types]
#0 10.61       14584 |       __pyx_v_cnames = bcf_index_seqnames(__pyx_v_self->hidx, __pyx_v_self->hdr, (&__pyx_v_n));
#0 10.61             |                      ^
#0 10.61       cyvcf2/cyvcf2.c:14667:22: warning: assignment to ‘char **’ from incompatible pointer type ‘const char **’ [-Wincompatible-pointer-types]
#0 10.61       14667 |       __pyx_v_cnames = tbx_seqnames(__pyx_v_self->idx, (&__pyx_v_n));
#0 10.61             |                      ^
#0 10.61       In file included from /usr/local/include/python3.9/pytime.h:6,
#0 10.61                        from /usr/local/include/python3.9/Python.h:81,
#0 10.61                        from cyvcf2/cyvcf2.c:96:
#0 10.61       cyvcf2/cyvcf2.c: In function ‘__pyx_f_5numpy_PyDataType_SHAPE’:
#0 10.61       cyvcf2/cyvcf2.c:46268:39: error: ‘PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘subarray’
#0 10.61       46268 |     __Pyx_INCREF(((PyObject*)__pyx_v_d->subarray->shape));
#0 10.61             |                                       ^~
#0 10.61       /usr/local/include/python3.9/object.h:112:41: note: in definition of macro ‘_PyObject_CAST’
#0 10.61         112 | #define _PyObject_CAST(op) ((PyObject*)(op))
#0 10.61             |                                         ^~
#0 10.61       cyvcf2/cyvcf2.c:1962:27: note: in expansion of macro ‘Py_INCREF’
#0 10.61        1962 |   #define __Pyx_INCREF(r) Py_INCREF(r)
#0 10.61             |                           ^~~~~~~~~
#0 10.61       cyvcf2/cyvcf2.c:46268:5: note: in expansion of macro ‘__Pyx_INCREF’
#0 10.61       46268 |     __Pyx_INCREF(((PyObject*)__pyx_v_d->subarray->shape));
#0 10.61             |     ^~~~~~~~~~~~
#0 10.61       cyvcf2/cyvcf2.c:46269:36: error: ‘PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘subarray’
#0 10.61       46269 |     __pyx_r = ((PyObject*)__pyx_v_d->subarray->shape);
#0 10.61             |                                    ^~
#0 10.61       error: command '/usr/bin/gcc' failed with exit code 1
#0 10.61       [end of output]
#0 10.61
#0 10.61   note: This error originates from a subprocess, and is likely not a problem with pip.
#0 10.61   ERROR: Failed building wheel for cyvcf2
#0 10.61   Running setup.py clean for cyvcf2
#0 10.73   DEPRECATION: Building 'vcftoolbox' using the legacy setup.py bdist_wheel mechanism, which will be removed in a future version. pip 25.3 will enforce this behaviour change. A possible replacement is to use the standardized build interface by setting the `--use-pep517` option, (possibly combined with `--no-build-isolation`), or adding a `pyproject.toml` file to the source tree of 'vcftoolbox'. Discussion can be found at https://github.com/pypa/pip/issues/6334
#0 10.73   Building wheel for vcftoolbox (setup.py): started
#0 10.86   Building wheel for vcftoolbox (setup.py): finished with status 'done'
#0 10.86   Created wheel for vcftoolbox: filename=vcftoolbox-1.5-py3-none-any.whl size=15672 sha256=c9cbe67770b2e6b57303ce357cefeb45e3fec4af60cffe10856d9889693a1483
#0 10.86   Stored in directory: /tmp/pip-ephem-wheel-cache-5sxzt9kf/wheels/a5/91/9f/01b01c386db3654216e5669e8f93b1c058ed370c51a855a1b0
#0 10.86 Successfully built vcftoolbox
#0 10.86 Failed to build cyvcf2
#0 10.92 error: failed-wheel-build-for-install
#0 10.92
#0 10.92 × Failed to build installable wheels for some pyproject.toml based projects
#0 10.92 ╰─> cyvcf2
------
failed to solve: executor failed running [/bin/sh -c pip install -r requirements.txt --no-cache-dir]: exit code: 1
(loqusdbapi) chiararasi@n132-p173 loqusdbapi %
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 ## [unreleased]
 ### Added
 - MIT license
+### Fixed
+- Docker compose image build on Apple Silicon by forcing x86_64 platform
 
 ## [0.1.14]
 ### Fixed

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -3,6 +3,7 @@ version: '2'
 services:
   loqusdbapi:
     build: .
+    platform: linux/amd64 # forces x86_64, avoids cyvcf2 ARM issues
     ports:
       - 9000:8000
     links:

